sound/soc/sof/core.c

```diff
@@ -310,6 +310,7 @@ int snd_sof_device_probe(struct device *dev, struct snd_sof_pdata *plat_data)
 	/* check all mandatory ops */
 	if (!sof_ops(sdev) || !sof_ops(sdev)->probe || !sof_ops(sdev)->run ||
 	    !sof_ops(sdev)->block_read || !sof_ops(sdev)->block_write ||
+	    !sof_ops(sdev)->mailbox_read || !sof_ops(sdev)->mailbox_write ||
```

Comment:
AMD platform is not using mmaped IO for IPC so make it an optional ops() callback and not a mandatory one.

Reply:
Not following @ajitkupandey. This patch precisely enables you to not use mmap IO but the mechanism of your choice.

Reply:
yeah i saw we can implement our custom mailbox_read()/write() but we aren't using mailbox read and write at the moment. If we make it mandatory we will be forced to implement such callbacks as directly sof_mailbox_read() not going to work for us so we prefer to make it an optional dsp ops () callback for now.

Reply:
@ajitkupandey OK. I think for now we can make this ops optional.

Reply:
@dbaluta @plbossart Thanks.. FYI we will implement custom mailbox_read()/write()  for AMD platform to make our code more streamline with SOF core but for now keeping these ops optional is really helpful.